pkg/kubelet/pleg/evented.go

```diff
@@ -270,7 +270,10 @@ func (e *EventedPLEG) processCRIEvents(containerEventsResponseCh chan *runtimeap
 			}
 			shouldSendPLEGEvent = true
 		} else {
-			if e.cache.Set(podID, status, err, time.Unix(0, event.GetCreatedAt())) {
+			// Filter some irrelevant Sandbox events to prevent the container lifecycle processing from being triggered by mistake.
+			// Not adding this code will not cause the e2e test to fail,
+			// but it will cause abnormal pod status in parallel testing and pod deletion.
```

Comment:
I guess #123087 was caused at staring a pod because a sandbox event woke up a pod worker and the worker saw a `created` container status.

Reply:
This should be relevant because the container runtime reports the same events for sandbox containers and regular containers, and the kubelet has never dealt with these issues. In fact, after the evented PLEG chooses to directly pass the events to the container lifecycle  (which is what the current PR does) instead of using the cache update mechanism this problem will occur even more frequently.

Reply:
Therefore, I add a judgment here to ensure that before the creation of regular containers, the creation events of sandbox containers will not trigger the lifecycle processing, and after the deletion of regular containers, the deletion events of sandbox containers will not trigger the lifecycle processing either. However, the cache will still be updated to the latest state.